### PR TITLE
refactor: simplify setup of JS bindings for async ops

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit greatly simplifies how ops are bound into JS functions.

When a new context if created, for each op we're binding it to
`Deno.core.ops[<name>]` - additionally doing a set up for async ops
(by calling `Deno.core.setUpAsyncStub` that generates an optimized
version of an op, with promise accounting and error mapping).

Starting with this commit `00_primordials.js` and `00_infra.js` are
executed in `bindings::initialize_context`, instead of `JsRuntime::init_extension_js`.
These two files are necessary to do anything useful in `deno_core` -
including executing `01_core.js` (which relies on ops heavily).

String codegen was removed from `bindings::initialize_context` and replaced
with a direct for-loop in Rust that calls V8 APIs.

All the changes related to how context is initialized result in 0.2ms startup
time improvement in `fs_module_loader` example:
```
hyperfine --warmup 30 -S none './fs_module_loader_new empty.js' './fs_module_loader empty.js'
Benchmark 1: ./fs_module_loader_new empty.js
  Time (mean ± σ):       6.2 ms ±   0.2 ms    [User: 3.6 ms, System: 1.5 ms]
  Range (min … max):     5.9 ms …   8.2 ms    470 runs

  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet PC without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.

Benchmark 2: ./fs_module_loader empty.js
  Time (mean ± σ):       6.4 ms ±   0.2 ms    [User: 3.8 ms, System: 1.5 ms]
  Range (min … max):     6.1 ms …   8.1 ms    475 runs

  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet PC without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.

Summary
  './fs_module_loader_new empty.js' ran
    1.03 ± 0.06 times faster than './fs_module_loader empty.js'

```

Closes https://github.com/denoland/deno_core/issues/445

### CODE REVIEW

**Comment by bartlomieju** [comment]
Add a test that checks async stack trace with an async op - `ErrorCaptureStackTrace` setup is funky

File: core/00_infra.js
Comment (bartlomieju) [inline core/00_infra.js]: `function asyncStub` is gone now

File: core/00_infra.js
Comment (bartlomieju) [inline core/00_infra.js]: Remove before landing

File: core/runtime/bindings.js
Comment (bartlomieju) [inline core/runtime/bindings.js]: All of this was dead code

File: core/runtime/jsruntime.rs
Comment (bartlomieju) [inline core/runtime/jsruntime.rs:1049]: This is a TODO, so probably don't want to land it :)

File: core/runtime/bindings.rs
Comment (bartlomieju) [inline core/runtime/bindings.rs:248]: No more funky codegen :D

File: core/runtime/bindings.rs
Comment (bartlomieju) [inline core/runtime/bindings.rs:247]: Add comment what's going on in this function.

Also this function should be combined with `initialize_deno_namespace_object` and execute `00_primordials.js` and `00_infra.js`.

File: core/01_core.js
Comment (bartlomieju) [inline core/01_core.js:274]: Sweet, this is now doing almost nothing

File: core/runtime/bindings.js
Comment (bartlomieju) [inline core/runtime/bindings.js:1]: Bye, bye

File: core/runtime/bindings.rs
Comment (bartlomieju) [inline core/runtime/bindings.rs:182]: No more `Deno.core.asyncOps`

File: core/runtime/jsrealm.rs
Comment (bartlomieju) [inline core/runtime/jsrealm.rs:261]: This should probably be moved to `FastString`?

**Review by mmastrac — APPROVED** [review]
LGTM assuming deno tests pass

**Pull request merged**
